pkg/domain/subscribers/handler.go

```diff
@@ -0,0 +1,60 @@
+package subscribers
+
+//go:generate mockgen -source=handler.go -destination=mock.go
+
+import (
+	"errors"
+	"github.com/gin-gonic/gin"
+	"net/http"
+	"se-school-case/pkg/domain"
+	"se-school-case/pkg/model"
+	"se-school-case/pkg/util/app-error"
+)
+
+type SubscriberInterface interface {
+	Add(email string) error
+	GetAll() ([]model.User, error)
+}
+
+type Handler struct {
+	subscriberService SubscriberInterface
+}
+
+func NewHandler(subscriberService SubscriberInterface) domain.Registrable {
```

Comment:
can we return structure instead of interface ? It would be more idiomatically https://duncanleung.com/go-idiom-accept-interfaces-return-types/